src/components/InstructionButton.js

```diff
@@ -1,39 +1,36 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
+import {withRouter} from 'react-router';
 import Button from 'react-bootstrap/lib/Button';
-import withStyles from 'isomorphic-style-loader/lib/withStyles';
+import useStyles from 'isomorphic-style-loader/useStyles';
 import styles from './InstructionButton.scss';
 import LinkContainer from 'react-router-bootstrap/lib/LinkContainer';
 import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 import {getTranslator} from '../selectors/translate';
 import {getLessonPath} from '../resources/lessonFrontmatter';
 
-class InstructionButton extends React.PureComponent {
-  handleKeyPress = () => this.context.router.push(this.props.path);
-
-  render() {
-    const {isReadme, onlyIcon, insideLink, path, buttonText} = this.props;
-    const options = {
-      className: onlyIcon ? styles.buttonOnlyIcon : styles.button,
-      bsStyle: 'guide',
-      bsSize: onlyIcon ? 'xs' : 'small',
-      componentClass: insideLink ? 'div' : 'a',
-      tabIndex: '0',
-      'aria-label': buttonText,
-      onKeyPress: this.handleKeyPress,
-    };
-    return (path ?
-      <LinkContainer to={path}>
-        <Button  {...options}>
-          <Glyphicon className={styles.icon} glyph={isReadme ? 'education' : 'pencil'}/>
-          <span className={onlyIcon ? '' : styles.textMargin}>{onlyIcon ? '' : buttonText}</span>
-        </Button>
-      </LinkContainer> :
-      null
-    );
-  }
-}
+const InstructionButton = ({isReadme, onlyIcon, insideLink, router, path, buttonText}) => {
+  useStyles(styles);
+  const options = {
+    className: onlyIcon ? styles.buttonOnlyIcon : styles.button,
+    bsStyle: 'guide',
+    bsSize: onlyIcon ? 'xs' : 'small',
+    componentClass: insideLink ? 'div' : 'a',
+    tabIndex: '0',
+    'aria-label': buttonText,
+    onKeyPress: () => router.push(path),
```

Comment:
I alle eksemplene jeg ser på nett, så står det at `withRouter` gir `history`, `location` og `match`, men det står ingenting om `router`. Er det et alias for `history`?

Reply:
Brukte dette svaret her... https://stackoverflow.com/a/40890372

Reply:
Er det ikke mulig å bruke browserHistory som jeg foreslo i noen commits tilbake?